osi_detectedlandmark.proto

```diff
@@ -225,17 +278,68 @@ message DetectedTrafficLight
     // and a high value indicates strong confidence.
     //
     optional double existence_probability = 8;
-    
+
     // A list of sensors which detected this detected entity.
     //
-    // If SensorData has detected entities and all detections are missing, then 
-    // e.g. the number of sensors can confirm the existence_probability.
-    // 
+    // If \c SensorData has detected entities and all detections are missing,
+    // then e.g. the number of sensors can confirm the
+    // \c #existence_probability.
+    //
     // \note This information can be determined via the detected entities'
     // detections ( \c ...Detection::object_id = 'this detected entity' ) and
     // the sensors (their IDs) to which these detections belong.
     //
-    repeated Identifier sensor_id = 10;    
+    // \note OSI uses singular instead of plural for repeated field names.
+    //
+    repeated Identifier sensor_id = 10;
+
+    //
+    // \brief Candidates for a detected traffic light as estimated by the
+    // sensor.
+    //
+    message EstimatedTrafficLight
+    {
+        // A list of candidates for this traffic light as estimated by the
+        // sensor.
+        //
+        // \note OSI uses singular instead of plural for repeated field names.
+        //
+        repeated CandidateTrafficLight candidate = 1;
```

Comment:
We cannot do:
optional TrafficLight traffic_light = 1; 
and at the same time:
repeated CandidateTrafficLight candidate = 1;

The naming convention must always follow the same pattern!
repeated CandidateTrafficLight candidate_traffic_lights = 1;

Reply:
One option for a pattern is, to use the message name as field name. (drawback: if you use a message twice in a message the field names would be the same. There are other drawbacks..).

Second option: you choose a field name which is clearly in the message context and you would not repeat infromation several times. 

In OSI we have a mix of both...